### PR TITLE
String tensors in tf.Graph decoder

### PR DESCRIPTION
**Pull request by popovaan**
Description: 
Implemented passing of string tensors to TF FE using vector of string.
Added test.

Ticket: 113210

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR - N/A
* [x]  Transformation preserves original framework node names - N/A
* [x]  Transformation preserves tensor names - N/A

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests - N/A
* [x]  Model Optimizer IR Reader check - N/A

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py [hunk 1]
```diff
@@ -587,6 +587,27 @@ def __call__(self, input1):
     return model, model_ref, {'example_input': example_input}
 
 
+def create_keras_layer_with_string_tensor(tmp_dir):
+    import tensorflow as tf
+
+    x = tf.keras.Input(shape=[1], name="input", dtype=tf.dtypes.int64)
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:593]:
```suggestion
    x = tf.keras.Input(shape=[1], name="input", dtype=tf.int64)
```

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py [hunk 2]
```diff
@@ -587,6 +587,27 @@ def __call__(self, input1):
...
+    var = tf.Variable("Text_1")
+    const = tf.constant("Text_2")
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:595]: please set dtype explicitly

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py [hunk 3]
```diff
@@ -587,6 +587,27 @@ def __call__(self, input1):
...
+    hash = tf.strings.to_hash_bucket([const, var], 3)
+
+    model = tf.keras.Model(inputs=[x], outputs=[hash + x])
+    example_input = tf.constant(3)
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:599]: it seems that `example_input` is not needed here.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py [hunk 4]
```diff
@@ -587,6 +587,27 @@ def __call__(self, input1):
...
+
+    param = ov.opset8.parameter([], dtype=np.int32)
+    conv = ov.opset8.convert(param, np.int64)
+    const = ov.opset8.constant([0, 2], dtype=np.int64)
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:603]: I am not sure that we test the required logic. I propose to create a graph with two sub-graphs: 1. Add(param1_float, param2_float) 2. Const(string)

We have transformation removing Const->Result this way the resulted graph will be with Add operation and two flow params

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py [hunk 5]
```diff
@@ -44,26 +46,47 @@ def tf_attr_to_ov(attr):
     return OVAny(tf_attr_to_numpy(attr))
 
 
+def try_decode(line: bytes):
+    assert isinstance(line, bytes), "Expected bytes, got object of type {}".format(type(line))
+    try:
+        return True, line.decode()
+    except ValueError:
+        return False, line
```
Comment (rkazants) [inline src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:54]: why do we need this logic? why can't we decode each string tensor element more universally?

Comment (popovaan) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:54]: This logic is needed to decode bytes to string. Bytes in general can fail to decode, if it is not a string or there is some unknown encoding. For this reason try-catch is used.
If we fail to decode it, then list of bytes objects will be returned from decoder.

What do mean by decoding each string tensor element more universally?

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py (hunk 2 above)
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py]: Done.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py (hunk 1 above)
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py]: Applied.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py (hunk 4 above)
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py]: Added test with such graph.

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py (hunk 5 above)
Comment (popovaan) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:54]: In offline discussion it was decided to replace `decode()` with `str()`. Done.

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py [hunk 6]
```diff
@@ -44,26 +46,45 @@ def tf_attr_to_ov(attr):
...
```
Comment (rkazants) [inline src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py:54]: agreed to remove it and do not change bytestream

**Review by rkazants — APPROVED** [review]
LGTM, please remove decoding for elements of string tensor

File: src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py (hunk 6 above)
Comment (popovaan) [reply src/bindings/python/src/openvino/frontend/tensorflow/node_decoder.py]: Removed.

**Pull request merged**